hyperspy/model.py

```diff
@@ -499,133 +378,6 @@ def _touch(self):
 
     __touch = _touch
 
-    def set_convolution_axis(self):
-        """
-        Creates an axis to use to generate the data of the model in the precise
-        scale to obtain the correct axis and origin after convolution with the
-        lowloss spectrum.
-        """
-        ll_axis = self.low_loss.axes_manager.signal_axes[0]
-        dimension = self.axis.size + ll_axis.size - 1
-        step = self.axis.scale
-        knot_position = ll_axis.size - ll_axis.value2index(0) - 1
-        self.convolution_axis = generate_axis(self.axis.offset, step,
-                                              dimension, knot_position)
-
-    def _connect_parameters2update_plot(self):
-        if self._plot_active is False:
-            return
-        for i, component in enumerate(self):
-            component.connect(
-                self._model_line.update)
-            for parameter in component.parameters:
-                parameter.connect(self._model_line.update)
-        if self._plot_components is True:
-            self._connect_component_lines()
-
-    def _disconnect_parameters2update_plot(self):
-        if self._model_line is None:
-            return
-        for component in self:
-            component.disconnect(self._model_line.update)
-            for parameter in component.parameters:
-                parameter.disconnect(self._model_line.update)
-        if self._plot_components is True:
-            self._disconnect_component_lines()
-
-    def as_signal(self, component_list=None, out_of_range_to_nan=True,
```

Comment:
Yes - I agree in principle. I moved these to Model1D and 2D because the disconnecting-connecting methods are not implemented for 2D but on reflection it doesn't really matter because it will just raise a NotImplementedError for Model 2D anyway -- I'll move the above bits to BaseModel